engine/runtime-with-polyglot/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala

```diff
@@ -2245,6 +2245,105 @@ class RuntimeVisualizationsTest
   }
 
+  it should "run visualisation default preprocessor" in {
+    val contextId       = UUID.randomUUID()
+    val requestId       = UUID.randomUUID()
+    val visualisationId = UUID.randomUUID()
+    val moduleName      = "Enso_Test.Test.Main"
+    val metadata        = new Metadata
+
+    val idMain = metadata.addItem(47, 6)
+
+    val code =
+      """import Standard.Visualization
+        |
+        |main =
+        |    fn = x -> x
+        |    fn
+        |""".stripMargin.linesIterator.mkString("\n")
+    val contents = metadata.appendToCode(code)
+    val mainFile = context.writeMain(contents)
+
+    val visualisationModule   = "Standard.Visualization.Preprocessor"
+    val visualisationFunction = "default_preprocessor"
+
+    // create context
+    context.send(Api.Request(requestId, Api.CreateContextRequest(contextId)))
+    context.receive shouldEqual Some(
+      Api.Response(requestId, Api.CreateContextResponse(contextId))
+    )
+
+    // Open the new file
+    context.send(
+      Api.Request(Api.OpenFileNotification(mainFile, contents))
+    )
+    context.receiveNone shouldEqual None
+
+    // push main
+    val item1 = Api.StackItem.ExplicitCall(
+      Api.MethodPointer(moduleName, moduleName, "main"),
+      None,
+      Vector()
+    )
+    context.send(
+      Api.Request(requestId, Api.PushContextRequest(contextId, item1))
+    )
+    val pushContextResponses =
+      context.receiveNIgnorePendingExpressionUpdates(3)
+    pushContextResponses should contain theSameElementsAs Seq(
+      Api.Response(requestId, Api.PushContextResponse(contextId)),
+      TestMessages.update(
+        contextId,
+        idMain,
+        ConstantsGen.FUNCTION
+      ),
+      context.executionComplete(contextId)
+    )
+
+    // attach visualisation
+    context.send(
+      Api.Request(
+        requestId,
+        Api.AttachVisualisation(
+          visualisationId,
+          idMain,
+          Api.VisualisationConfiguration(
+            contextId,
+            Api.VisualisationExpression.ModuleMethod(
+              Api.MethodPointer(
+                visualisationModule,
+                visualisationModule,
+                visualisationFunction
+              ),
+              Vector()
+            )
+          )
+        )
+      )
+    )
+    val attachVisualisationResponses =
+      context.receiveNIgnoreExpressionUpdates(2)
+    attachVisualisationResponses should contain(
+      Api.Response(requestId, Api.VisualisationAttached())
+    )
+    val Some(data) = attachVisualisationResponses.collectFirst {
+      case Api.Response(
+            None,
+            Api.VisualisationUpdate(
+              Api.VisualisationContext(
+                `visualisationId`,
+                `contextId`,
+                `idMain`
+              ),
+              data
+            )
+          ) =>
+        data
+    }
+    val stringified = new String(data)
+    stringified shouldEqual "\"Function\""
```

Comment:
Do you think we could make `Function.to_display_text = Function.to_text`?

The `to_text` provides much nicer information about the origin of the closure, like:
```
> f = x-> f = x-> x+1
>>> Nothing
> f
>>> f[<interactive_source>:1-2]
```

(this is from REPL, in source files this would be a path to the file)

Reply:
We already use `to_display_text` as a "short" visualization of error values. This implementation is consistent with it.
For the proper function visualization I'd rather implement a custom `to_default_visualization_data` method.

Reply:
OK, sure